### PR TITLE
Implement haskell-goto-first-error

### PR DESCRIPTION
**Pull request by bergey**
Supercedes https://github.com/haskell/haskell-mode/pull/818

### CODE REVIEW

File: tests/haskell-load-tests.el [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+;;; Code:
+
+(require 'ert)
+(require 'haskell-test-utils)
+
+(require 'haskell-load)
+
+(defun insert-errors ()
+  (insert "import           Control.Applicativ\nimport           Data.Mayb\nimport           Data.String")
+  (let ((a (make-overlay 18 36))
```
Comment (gracjan) [inline tests/haskell-load-tests.el]: Please use `re-search-forward` and then `(match-begin 0)` and `(match-end 0)` to put the overlays. Should somebody come and need to change these tests they will be intimidated by character counting and won't do it.

File: tests/haskell-load-tests.el [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 1)
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
+
+(ert-deftest goto-first-error-after ()
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 92)
```
Comment (gracjan) [inline tests/haskell-load-tests.el]: Please use `re-search-forward` instead of `goto-char`.

File: tests/haskell-load-tests.el [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
+
+(ert-deftest goto-first-error-between ()
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 37)
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
```
Comment (gracjan) [inline tests/haskell-load-tests.el]: Please either use `re-search-forward` or use `looking-at-p` to check if we are in the correct spot in the source file.

**Comment by gracjan** [comment]
Very nice! Please get rid of those hardcoded numbers and then we are good to merge.

**Comment by bergey** [comment]
@gracjan I force pushed the changes you requested.

File: tests/haskell-load-tests.el [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+;;; Code:
+
+(require 'ert)
+(require 'haskell-test-utils)
+
+(require 'haskell-load)
+
+(defun insert-errors ()
+  (insert "import           Control.Applicativ\nimport           Data.Mayb\nimport           Data.String")
```
Comment (gracjan) [inline tests/haskell-load-tests.el]: One more nitpick: Can I have one space here between `import` and module names?

Counting spaces might be intimidating for future developers, too.

**Comment by bergey** [comment]
Makes sense.  I started from real Haskell code, and the extra spaces are what `stylish-haskell` does.  But they do make the tests harder to read otherwise.  I force pushed again.

**Comment by gracjan** [comment]
Perfect, thanks.

**Pull request merged**